### PR TITLE
chore: add a script to help generate basic release notes

### PR DESCRIPTION
**Pull request by hamidzr**
## Test Plan

no tests needed. 

## Commentary (optional)

I forget the standard titles, prefixes, and rst format and it adds friction to adding release notes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by tara-det-ai — COMMENTED** [review]
i added suggested reviewers

File: docs/release-notes/generate-release-note.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+categories=("Bug Fixes" "Security Fixes" "Breaking Changes" "Improvements" "New Features")
+components=("WebUI" "Notebook" "TensorBoard" "Command" "Shell" "Experiment" "API" "Images")
+
+autocomplete() {
```
Comment (dzhu) [inline docs/release-notes/generate-release-note.sh:6]: This doesn't appear to actually allow for a write-in entry, and attempting to put one in causes "Invalid selection" and a newline to be included in the output. Looks like you want to examine `$REPLY` in the `else` branch. Also `autocomplete` is not a great name.

File: docs/release-notes/generate-release-note.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    done
+}
+
+echo "Enter the filename for the release note (without extension):"
+read filename
+
+echo "Choose a category (or enter a new one):"
+category=$(autocomplete "${categories[@]}")
+
+echo ":orphan:" >"${filename}.rst"
```
Comment (dzhu) [inline docs/release-notes/generate-release-note.sh:24]: Maybe always put the file in the right directory (roughly `dirname $0`). Might also be nice to have a variable for the full output path.

File: docs/release-notes/generate-release-note.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    echo "-  ${component}: ${details}" >>"${filename}.rst"
+
+    echo "Do you want to add another entry? (yes/no)"
+    read answer
+    if [[ $answer != "yes" ]]; then
+        break
+    fi
+done
+
```
Comment (dzhu) [inline docs/release-notes/generate-release-note.sh:44]: Apply `rstfmt` when done.

File: docs/release-notes/generate-release-note.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
```
Comment (dzhu) [inline docs/release-notes/generate-release-note.sh:2]: `set -eu`

**Review by dzhu — COMMENTED** [review]
This is past the point where I would switch to Python, but if you're maintaining it, I'm not about to complain.

**Comment by hamidzr** [comment]
> This is past the point where I would switch to Python,

I agree it started out smaller..

I'll close and revisit once I join back the core team so I can dogfood it better.

**Pull request closed without merging**